### PR TITLE
Move the header files in the AWS Core gem based on the latest gem structure guideline

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

- Passed unit tests and end-to-end tests on Windows
- Tested with the AutomatedTesting project on Linux

### CODE REVIEW

File: Gems/AWSCore/Code/CMakeLists.txt [hunk 1]
```diff
@@ -16,10 +16,10 @@ ly_add_target(
          ${pal_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
     INCLUDE_DIRECTORIES
         PUBLIC
-            Include/Public
```
Comment (onecent1101) [inline Gems/AWSCore/Code/CMakeLists.txt:19]: minor: extra slash

File: Gems/AWSCore/Code/CMakeLists.txt [hunk 2]
```diff
@@ -68,10 +68,10 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
             ${pal_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_editor_files.cmake
         INCLUDE_DIRECTORIES
             PRIVATE
-                Include/Private
+                Source
                 ${pal_dir}
             PUBLIC
-                Include/Public
```
Comment (onecent1101) [inline Gems/AWSCore/Code/CMakeLists.txt:74]: Minor: extra slash

File: Gems/AWSCore/Code/CMakeLists.txt [hunk 3]
```diff
@@ -144,8 +144,8 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
             awscore_tests_files.cmake
         INCLUDE_DIRECTORIES
             PRIVATE
-                Include/Private
-                Include/Public
+                Source
+                Include/
```
Comment (onecent1101) [inline Gems/AWSCore/Code/CMakeLists.txt:148]: Minor: extra slash

File: Gems/AWSCore/Code/CMakeLists.txt [hunk 4]
```diff
@@ -168,9 +168,9 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
                 ${pal_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_editor_tests_files.cmake
             INCLUDE_DIRECTORIES
                 PRIVATE
-                    Include/Private
+                    Source
                     ${pal_dir}
-                    Include/Public
+                    Include/
```
Comment (onecent1101) [inline Gems/AWSCore/Code/CMakeLists.txt:173]: minor: extra slash

**Review by onecent1101 — APPROVED** [review]

**Comment by junbo75** [comment]
Merged from latest dev and resolved conflicts on https://github.com/o3de/o3de/blob/development/Gems/AWSCore/Code/CMakeLists.txt

**Comment by allisaurus** [comment]
Changes LGTM, but can you please indicate what testing you did to validate these changes? (e.g., editor builds, unit tests pass, e2e tests pass)

**Comment by junbo75** [comment]
> Changes LGTM, but can you please indicate what testing you did to validate these changes? (e.g., editor builds, unit tests pass, e2e tests pass)

Yeah, just updated the description.

**Review by allisaurus — APPROVED** [review]

**Pull request merged**